app/content-panel/blogs/[id]/page.tsx

```diff
@@ -0,0 +1,95 @@
+'use client';
+
+import React, { useState, useEffect } from 'react';
+import styles from './BlogDetailsPage.module.scss';
+import BlogDetailsCard from '../../../../components/reusable-components/blogDetails-card/BlogDetailsCard';
+
+const BlogDetailsPage = ({ params }: { params: { id: string } }) => {
+  const [isEditable, setIsEditable] = useState(false);
+  const [selectedImage, setSelectedImage] = useState<string | null>(null);
+  const [title, setTitle] = useState('N/A');
+  const [slug, setSlug] = useState('N/A');
+  const [image, setImage] = useState('');
+  const [author, setAuthor] = useState({ first_name: 'N/A', last_name: 'N/A' });
+  const [publishDate, setPublishDate] = useState('N/A');
+  const [tags, setTags] = useState<string[]>([]);
```

Comment:
You can reduce the number of state variables and combine them in one state object called blogDetails

Reply:
You can maybe keep the image state separate if you want to show a preview of the uploaded image

Reply:
single state object created